core/validatorapi/router.go

```diff
@@ -86,142 +86,172 @@ func NewRouter(ctx context.Context, h Handler, eth2Cl eth2wrap.Client) (*mux.Rou
 		Name    string
 		Path    string
 		Handler handlerFunc
+		Methods []string
```

Comment:
why a slice rather than a string? In this case, we will need to write a single handler which will handle all the http statuses. But if specify just one method per endpoint then we can separate handlers for each method which might be more clean than the current approach. WDYT?

Reply:
Generally when designing APIs like this I found that one might want to handle the same endpoint in different HTTP methods.